### PR TITLE
Check in/67398/unit test coverage

### PR DESCRIPTION
**Pull request by brianseek**
## Summary

This ticket is to increase our unit test coverage in the `Functions` category specifically. This increases the coverage from 77.03% to 84.94%. In some areas I ignored coverage for things like mocks and tests. This also addresses one flaky test, see details here: https://docs.google.com/spreadsheets/d/1tqsbBonIbSSq3enPQcd-yg0KSzSBp38TRD3D_tKq3qU/edit#gid=0

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#67398

## Testing done

PR is all tests.

## Screenshots

Screenshot 2023-10-12 at 9 15 27 AM

## What areas of the site does it impact?

Unit tests for check-in app

### CODE REVIEW

**Review by wullaski — APPROVED** [review]
LGTM!

File: src/applications/check-in/pre-check-in/pages/Demographics/tests/Demographics.unit.spec.jsx [hunk 1]
```diff
@@ -93,5 +93,28 @@ describe('pre-check-in', () => {
...
+    it('has a clickable no button', () => {
+      const component = render(
+        <CheckInProvider store={{ veteranData }}>
+          <Demographics />
+        </CheckInProvider>,
+      );
+
+      expect(component.getByText('Is this your current contact information?'))
+        .to.exist;
+      component.getByTestId('no-button').click();
```
Comment (lee-delarm6) [inline src/applications/check-in/pre-check-in/pages/Demographics/tests/Demographics.unit.spec.jsx]: See if there's a better way to detect that click is here, OR that it works (and second section below here)

Comment (lee-delarm6) [reply src/applications/check-in/pre-check-in/pages/Demographics/tests/Demographics.unit.spec.jsx]: Resolved with added spy on click event

**Review by lee-delarm6 — APPROVED** [review]
Resolved with added click event spy 👍

**Pull request merged**